### DISCUSSION TITLE
Invalid Access Token

### DISCUSSION THREAD

**Discussion by yagodaoud** [spotify-web-api-java/spotify-web-api-java#346] [Q&A]
Even though I can get the accessToken and set it properly into the spotify api object, it's invalid somehow (even after just creating it).
in the spotify api connection class: 
```
private static final SpotifyApi spotifyApi = new SpotifyApi.Builder()
            .setClientId(clientId)
            .setClientSecret(clientSecret)
            .build();

    public static void clientCredentials_Async() {
        try {
            final ClientCredentialsRequest clientCredentialsRequest = spotifyApi.clientCredentials()
                    .build();

            final CompletableFuture<ClientCredentials> clientCredentialsFuture = clientCredentialsRequest.executeAsync();

            final ClientCredentials clientCredentials = clientCredentialsFuture.join();

            spotifyApi.setAccessToken(clientCredentials.getAccessToken());
            System.out.println(clientCredentials.getAccessToken());

            System.out.println("Expires in: " + clientCredentials.getExpiresIn());
```
until here the code works normally, but when trying to get a playlist by its id (the playlist is not private or has any typo in the id):
 ```
  private static final String playlistId = "5GOQSS6SUFDD6SyMfVRMA5";
    private static final SpotifyApi spotifyApi = SpotifyApiConnection.getSpotifyApi();
    private static final GetPlaylistRequest getPlaylistRequest = spotifyApi.getPlaylist(playlistId)
            .build();


    public static void getPlaylist_Async() {
        try {
            final CompletableFuture<Playlist> playlistFuture = getPlaylistRequest.executeAsync();

            final Playlist playlist = playlistFuture.join();

            System.out.println("Name: " + playlist.getName());
        } catch (CompletionException e) {
            System.out.println("Error: " + e.getCause().getMessage());
        } catch (CancellationException e) {
            System.out.println("Async operation cancelled.");
        }
    }

    public static void main(String[] args) throws InterruptedException {
        SpotifyApiConnection.clientCredentials_Async(); // Fetch the access token
        getPlaylist_Async();
    }
}
 ```
 The output is: BQAwiNqNEe1lz_wkBFNBEydg4telS29eViMt9KQR-yxZA_cbuqaEEdcTymrVAP-injggJP-3QoUEUBByY6AnJkRQUcmDK-sGKr9IcLlK4e9p-KvwDbA //the access token isn't null as you can see
Expires in: 3600
Error: Invalid access token

**Comment by dargmuesli** [comment]
This library does not interfere with the validity of what Spotify's API provides. If Spotify's API says the access token in invalid, you should check if the access token you use for the offending request is really the access token you should use.
For example, you could verify that `spotifyApi`, for which the access token appears to be set, is the same as what you get from `SpotifyApiConnection.getSpotifyApi()`.
Again, this is a question on how to use Spotify's API. This repository is just a single wrapper for it, multiple others exist, so maybe a StackOverflow post with the full (but minimal) reproduction reaches more people and a quicker solution :)
